dao/source_dao.go

```diff
@@ -17,13 +17,16 @@ var GetSourceDao func(*SourceDaoParams) SourceDao
 
 // getDefaultRhcConnectionDao gets the default DAO implementation which will have the given tenant ID.
 func getDefaultSourceDao(daoParams *SourceDaoParams) SourceDao {
-	var tenantID *int64
+	var tenantID, userID *int64
+
 	if daoParams != nil && daoParams.TenantID != nil {
```

Comment:
I would be ok with this panic-ing. We should _never ever_ pass in a daoParams struct with a nil tenant/user id.

Reply:
(1) It looks we can't now because of internal source list: https://github.com/RedHatInsights/sources-api-go/blob/ca719c20a38f5ad07adb4e33bdcd39f86a6e3718/internal_handlers.go 

(2) Anyway it would be probably better this way. Previously I was expecting that 
when `SourceDaoParams.Userid == nil` - we are filtering sources record with user_id = NULL
when `SourceDaoParams.Userid != nil` - we are filtering sources record with user_id = SourceDaoParams.Userid

But agree to have always populated `SourceDaoParams.Userid` and I will update it with this approach.

**Question** when we receive status message in status listener - can we rely that in header we will have also `Userid` in headers in received status message or  do we need to confirm this with teams who sends those message ?

Reply:
Update: 

> We should never ever pass in a daoParams struct with a nil tenant/user id.

To reach this we need to use string userid from xhr header - so I need to convert `SourceDaoParams.Userid` to string.

See comment here https://github.com/RedHatInsights/sources-api-go/pull/358#discussion_r918136834

or we can follow original pattern mentioned (2) and allow nil userid in SourceDaoParams struct - then we can do GetIdByUserID(from comment below) check in middleware.